packages/esm-patient-forms-app/src/offline.ts

```diff
@@ -73,3 +86,51 @@ async function syncPersonUpdate(personUuid?: string, personUpdate?: any) {
     body: personUpdate,
   });
 }
+
+export async function setupDynamicFormDataHandler() {
+  setupDynamicOfflineDataHandler({
+    id: 'esm-patient-forms-app:form',
+    type: 'form',
+    displayName: 'Patient forms',
+    async isSynced(identifier) {
+      const expectedUrls = await getCacheableFormUrls(identifier);
+      const absoluteExpectedUrls = expectedUrls.map((url) => window.origin + makeUrl(url));
+      const cache = await caches.open('omrs-spa-cache-v1');
+      const keys = (await cache.keys()).map((key) => key.url);
+      return absoluteExpectedUrls.every((url) => keys.includes(url));
```

Comment:
I am honestly torn about whether we should extract this logic here into a utility function in core. It is more or less the same code in every instance where it occurs (at the moment - it doesn't necessarily have to stay that way). I think it is duplicated around 5 times right now.
The reason for _not_ doing it at the moment is that _not_ moving it into core means that we don't have to commit to a new API that needs to be supported in the long run. Maybe it makes sense to wait and see how this develops and then, at some point, we can move it to core? Any thoughts?

Reply:
> [...] 3 times, its time to move it into a proper function [...]

That sounds like a good approach. I think in this specific case, I'd like to delay it a little bit to move things forward, but I'll keep it on my todo list (and perhaps even think about a proper way to do this - functions for cache interaction would generally be a nice addition to core, maybe even without an `@internal` modifier).